File: LeetSync/3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        max_len = 0
        left = 0
        count = {}

        for right, char in enumerate(s):
            count[char] = count.get(char, 0) + 1

            while count[char] > 2:
                count[s[left]] -= 1
                left += 1

            max_len = max(max_len, right - left + 1)

        return max_len

File: LeetSync/3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int maxLen = 0;
        int left = 0;
        int count[26] = {0};

        for (int right = 0; right < s.length(); ++right) {
            count[s[right] - 'a']++;

            while (count[s[right] - 'a'] > 2) {
                count[s[left] - 'a']--;
                left++;
            }

            maxLen = max(maxLen, right - left + 1);
        }

        return maxLen;
    }
};